agent/consul/server_test.go

```diff
@@ -1129,6 +1133,126 @@ func TestServer_RPC(t *testing.T) {
 	}
 }
 
+func TestServer_RPC_MetricsIntercept_Off(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	storage := make(map[string]float32)
+	keyMakingFunc := func(key []string) string {
+		return strings.Join(key, "+")
+	}
+
+	simpleRecorderFunc := func(key []string, val float32, labels []metrics.Label) {
+		storage[keyMakingFunc(key)] = val
+	}
+
+	_, conf := testServerConfig(t)
+
+	// "disable" metrics net/rpc interceptor
+	conf.GetNetRPCInterceptorFunc = nil
+
+	s, err := newServer(t, conf)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	defer s.Shutdown()
+	testrpc.WaitForTestAgent(t, s.RPC, "dc1")
+
+	// "hijack" the rpc recorder for asserts;
+	// note that there will be "internal" net/rpc calls made
+	// that will still show up; those don't go thru the net/rpc interceptor;
+	// see consul.agent.rpc.middleware.RPCTypeInternal for context
+	s.rpcRecorder = &middleware.RequestRecorder{
```

Comment:
Is there a data race here? I assume the test agent behind the scenes will be performing RPCs and swapping in an RPC recorder seems like it could cause the test to panic in the worst case.

Reply:
‼️ that is quite possible; I'm moving the `recorder` to be a part of deps too so this shouldn't be a problem anymore.